### ISSUE TITLE
Dropout keys cause compilation mismatches

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/axon#443]
Dropout keys are somehow getting converted from u32 to f32 in a template argument during training loop, which makes it impossible to run a training loop with strict compilation enabled (to avoid recompilations)
